packages/runtime/src/runtime/hooks/useEnsembleAction.tsx

```diff
@@ -205,18 +205,20 @@ export const useInvokeAPI: EnsembleActionHook<InvokeAPIAction> = (action) => {
   const { apis, setData } = useScreenData();
   const [isComplete, setIsComplete] = useState<boolean>();
   const [isLoading, setIsLoading] = useState<boolean>();
-
   const [context, setContext] = useState<unknown>();
-
   const evaluatedInputs = useEvaluate(action?.inputs, { context });
-  const onResponseAction = useEnsembleAction(action?.onResponse);
-  const onErrorAction = useEnsembleAction(action?.onError);
 
   const api = useMemo(
     () => apis?.find((model) => model.name === action?.name),
     [action?.name, apis],
   );
 
+  const onInvokeAPIResponseAction = useEnsembleAction(action?.onResponse);
+  const onInvokeAPIErrorAction = useEnsembleAction(action?.onError);
+
+  const onAPIResponseAction = useEnsembleAction(api?.onResponse);
+  const onAPIErrorAction = useEnsembleAction(api?.onError);
```

Comment:
Don't call hook multiple times per callback. Just get the correct callback model once.

Reply:
Discussed offline, both are needed if there are multiple onResponse for onLoad and on the API definition. However we still need to handle this for the code `invokeAPI` case.

Reply:
This applies to invokeAPI too. The API is only called when you use invokeAPI. Even during onLoad, invokeAPI is used, triggering the API in that case.

Reply:
I meant when you call ensemble.invokeAPI from code? It works there too?

Reply:
No, in `ensemble.invokeAPI` we are not passing any onResponse or onError callback. so it is not intended to work there

Reply:
Right, but when you call `ensemble.invokeAPI`, should the `onResponse` of the API model fire?

Reply:
No

Reply:
> Right, but when you call `ensemble.invokeAPI`, should the `onResponse` of the API model fire?

It should

Reply:
I'll add this in a separate PR